src/components/kubectl/proxy.ts

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+import * as vscode from 'vscode';
+import * as opn from 'opn';
+
+import {createReadStream, exists, open, unlink} from 'fs';
+import {resolve} from 'path';
+import {fs} from '../../fs';
+
+const KUBE_DASHBOARD_URL = "http://localhost:8001/ui/";
+const TERMINAL_NAME = "Kubernetes Proxy";
+const PROXY_OUTPUT_FILE = resolve(__dirname, 'proxy.out');
+
+// The instance of the terminal running Kubectl Proxy.
+let terminal:vscode.Terminal;
+
+/**
+ * Runs `kubectl proxy` in a terminal process spawned by the extension, and opens the Kubernetes
+ * dashboard in the user's preferred browser.
+ */
+export async function dashboardKubernetes ():Promise<void> {
+    // If we've already got a terminal instance, just open the dashboard.
+    if (terminal) {
+        opn(KUBE_DASHBOARD_URL);
+        return;
+    }
+
+    let outputExists;
+
+    try {
+        outputExists = await fs.existsAsync(PROXY_OUTPUT_FILE);
+        if (!outputExists) {
+            await fs.openAsync(PROXY_OUTPUT_FILE, 'w+');
+        }
+    } catch (e) {
+        vscode.window.showErrorMessage("Something went wrong when ensuring the Kubernetes API Proxy output stream existed");
+        return;
+    }
+
+    // Read kubectl proxy's stdout as a stream.
+    const proxyStream = createReadStream(
+        PROXY_OUTPUT_FILE,
+        {encoding: 'utf8'}
+    ).on('data', onStreamData);
+
+    terminal = vscode.window.createTerminal(TERMINAL_NAME);
+    vscode.window.onDidCloseTerminal(onClosedTerminal);
+
+    // stdout is also written to a file via `tee`. We read this file as a stream
+    // to listen for when the server is ready.
+    terminal.sendText(`kubectl proxy | tee ${PROXY_OUTPUT_FILE}`, true);
+    terminal.show();
```

Comment:
Is it worth passing `true` here so that the terminal doesn't steal focus?  It is kind of moot since we're about to open an enormous browser window which will _steal the heck_ out of the focus... but when the user Alt+Tabs back to VS Code, it might be good to have focus still on the editor?

Reply:
I think this is required to make the terminal run the command: https://code.visualstudio.com/docs/extensionAPI/vscode-api#Terminal, otherwise it waits for the user to do it themselves.

Reply:
I tried a trivial test and the command still seemed to be executed if you passed `true` for `preserveFocus`, at least on Windows.  Do you see something different?  If so, then I agree we can't pass the option.

Reply:
er derp; i thought this comment was with regards to the sendText call. I tried the change and it seems to work fine :)